include/z64map.h

```diff
@@ -1,6 +1,178 @@
 #ifndef Z64MAP_H
 #define Z64MAP_H
+#include "ultra64.h"
+#include "z64scene.h"
+
+struct PlayState;
+
+/*
+Handles L-minimap and pause screen dungeon minimap system.
+
+z_map_data.c stores MapSpriteInfo and helper functions to "simplify" accessing this data
+
+Two separate Id systems are used to access data
+
+mapId
+0x000-0x004 fetches L-minimap data from gameplay_dangeon_keep
+0x005-0x039 fetches dungeon minimap data from map_i_static
+0x100-0x161 fetches L-minimap data from map_grand_static
+
+mapCompactId
+0x000-0x039 fetches dungeon minimap data from map_i_static
+0x03A-0x09B fetches L-minimap data from map_grand_static
+*/
 
 #define FLOOR_INDEX_MAX 4
 
+/* z_map_data */
+#define MAPDATA_MAP_I_MAX 0x3A
+#define MAPDATA_MAP_GRAND 0x100
+#define MAPDATA_GET_MAP_GRAND_ID_FROM_COMPACT_ID(id) (id + (MAPDATA_MAP_GRAND - MAPDATA_MAP_I_MAX))
+#define MAPDATA_GET_MAP_GRAND_ID_FROM_MAP_ID(mapId) (mapId - MAPDATA_MAP_GRAND)
+
+#define MAPDATA_MID_GAMEPLAY_DANGEON_KEEP 0
+#define MAPDATA_MID_MAP_GRAND_STATIC 1
+#define MAPDATA_MID_MAP_I_STATIC 2
+
+#define MAPDATA_CPID_MAP_GRAND_STATIC 0
+#define MAPDATA_CPID_MAP_I_STATIC 1
+#define MAPDATA_CPID_2 2
+
+#define MAPDATA_DRAW_0 0
+#define MAPDATA_DRAW_1 1
+#define MAPDATA_DRAW_2 2
+#define MAPDATA_DRAW_3 3
+
+typedef struct {
+    /* 0x0 */ u8 texWidth;
+    /* 0x1 */ u8 texHeight;
+    /* 0x2 */ s16 offsetX;
+    /* 0x4 */ s16 offsetY;
+    /* 0x6 */ u8 drawType;
+    /* 0x7 */ u8 colorIndex;
+    /* 0x8 */ s16 scale;
+} MapSpriteInfo; // size = 0xA
```

Comment:
A lot of these structs in `z64map.h` seem local only to data in `map_disp.c`. Could you move these structs to `z_map_disp.c`?

Reply:
all structs are only used in either `map_data.c` or `map_disp.c`; should I move all the structs into their C files?

Reply:
I think that'd be fine yeah